doc/source/data/api/dataset.rst

```diff
@@ -149,3 +149,17 @@ Serialization
    Dataset.has_serializable_lineage
    Dataset.serialize_lineage
    Dataset.deserialize_lineage
+
+
+Internals
+---------
+
+.. autosummary::
+    :toctree: doc/
```

Comment:
Oh cool, this deprecates most of my fix PR here: https://github.com/ray-project/ray/pull/34228

Btw, is there a way to hide this internals section by default? Some of these are legacy backwards compatibility aliases that we don't want to expose.

Reply:
@ericl yes, toctrees can be `:hidden:`, if this is what we want.